File: SolutionDesigner/ClientApp/src/app/dashboard-item.ts
import { GridsterItem } from "angular-gridster2";
import { LineChartOptions } from "./configuration-panel/models/line-chart";
import { GaugeChartOptions } from "./configuration-panel/models/gauge-chart";
import { Column } from "./column";
import { TextElement } from "./text-element";

export interface DashboardItem extends GridsterItem {
  id: number;
  itemType: string;
  itemGroup: string;
  queries: Array<Query>;
//   cols: number;
//   rows: number;
//   x: number;
//   y: number;
}

export interface ChartDashboardItem extends DashboardItem {
    data: Array<any> | any;
    chartOptions: LineChartOptions | GaugeChartOptions;
    columns: Array<Column>;
}

// tslint:disable-next-line:no-empty-interface
export interface AppDashboardItem extends DashboardItem {
}

export interface TextDashboardItem extends DashboardItem {
  textOptions: TextElement;
}

export interface Query {
    guiItemName: string;
    query: string;
}

File: SolutionDesigner/Models/DashboardItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace SolutionDesigner.Models
{
    public class DashboardItem
    {
        [Key]
        public int? Id { get; set; }
        public string itemType { get; set; }
        public string itemGroup { get; set; }
        public List<Query> queries { get; set; }
        public int x { get; set; }
        public int y { get; set; }
        public int cols { get; set; }
        public int rows { get; set; }
    }
}


